### PR TITLE
Find the cheapest book and calculate the sum of digits from the number

### PR DESCRIPTION
**Pull request by Sasha1152**
Tasks #1 and #2

### CODE REVIEW

File: Oleksandr_Shmigelskiy/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+This is a branch of student Oleksandr.
+--------------------------------------
++ I am not professional programmer, but I want to be.
++ I like Python that is why I joined to this group.
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/README.md]: Please add your expectations and an ideal finishing of this learning.

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+"""
+Task #1
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: Docstrings should be self-descriptive and informative.

It should be something like:

"This module solves such a problem:
...
"""

Or something similar. You should be able to read it like a story, not like a copy-pasted task description.

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Calculate sum of all digits of a number set in string.
+Example:
+num = ‘12345’
+Your script should calculate it like: 1+2+3+4+5 = 15
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: Whose `your`?

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+num = ‘611091234512’
+result = 6+1+1+0+9+1+2+3+4+5+1+2 = 35
+"""
+
+
+def calculate_digits(num):
+    num = str(num)
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: Do we really need this?

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    sum_of_digits = 0
+    for digit in num:
+        try:
+            sum_of_digits += int(digit)
+        except ValueError:
+            continue
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: Well, what do you expect from `a5sd45g6`?

Why did you add this extra-behavior?

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+"""
+ Find the cheapest book
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task02_cheapest_book.py]: The same comment related to docstring as for the previous module.

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    ‘price’: 650
+    }
+]
+
+Your script should return: `The cheapest book is ‘Fluent Python’. It costs 650 grn.
+"""
+
+
+def get_cheapest_book(bookslist):
+    cheapest_book = bookslist[0]
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task02_cheapest_book.py]: What do you expect for an empty list of books?

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    for book in bookslist[1:]:
+        if book['price'] < cheapest_book['price']:
+            cheapest_book = book
+
+    return 'The cheapest book is "{0}". It costs {1} grn.'\
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task02_cheapest_book.py]: It's a bad idea to return formatted string.
It's much better to return the cheapest book and then format your final string in/before print.

File: Oleksandr_Shmigelskiy/README.md (hunk 1 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/README.md]: I altered it

File: Oleksandr_Shmigelskiy/task01_calc_digits.py (hunk 2 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task01_calc_digits.py]: fixed

File: Oleksandr_Shmigelskiy/task01_calc_digits.py (hunk 3 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task01_calc_digits.py]: deleted

File: Oleksandr_Shmigelskiy/task01_calc_digits.py (hunk 4 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task01_calc_digits.py]: of course, this operation expand the possibility of the function

File: Oleksandr_Shmigelskiy/task01_calc_digits.py (hunk 5 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task01_calc_digits.py]: this exception permit work with letters

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
+"""
+This function calculate sum of all digits of a number.
+A number can be represented in integer or string type.
+Value of the letters or any blank symbol equated to zero.
+
+Example:
+num = '12345'
+the function calculate it like: 1+2+3+4+5 = 15
+num = 'a5sd45g6'
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: I don't think we need this part here.

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+result = 5+4+5+6 = 20
+"""
+
+
+def calculate_digits(num):
+    if type(num) is int:
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: It's too complicated, really!

I suggest something similar to:

    result = sum(int(digit) for digit in number)

And that's it!

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py (hunk 6 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task02_cheapest_book.py]: deleted

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py (hunk 7 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task02_cheapest_book.py]: insert the empty list in this function has no sense

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py (hunk 8 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task02_cheapest_book.py]: fixed

File: Oleksandr_Shmigelskiy/task01_calc_digits.py (hunk 9 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task01_calc_digits.py]: deleted

File: Oleksandr_Shmigelskiy/task01_calc_digits.py (hunk 10 above)
Comment (Sasha1152) [reply Oleksandr_Shmigelskiy/task01_calc_digits.py]: done

File: Oleksandr_Shmigelskiy/task02_cheapest_book.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
+
+def get_cheapest_book(bookslist):
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task02_cheapest_book.py]: I don't understand why do you need this function here, but it might be...

File: Oleksandr_Shmigelskiy/task01_calc_digits.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    7
+    >>> calculate_digits('')
+    0
+    >>> calculate_digits('611091234512')
+    35
+    """
+
+    sum_of_digits = 0
+    for digit in num:
+        sum_of_digits += int(digit)
```
Comment (pivancy) [inline Oleksandr_Shmigelskiy/task01_calc_digits.py]: You're able to replace it just with `return sum(int(digit) for digit in num)`

**Comment by pivancy** [comment]
👍
